sdks/python/apache_beam/io/gcp/bigtable_io_write.py

```diff
@@ -0,0 +1,166 @@
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+"""BigTable connector
+
+This module implements writing to BigTable tables.
+The default mode is to set row data to write to BigTable tables.
+The syntax supported is described here:
+https://cloud.google.com/bigtable/docs/quickstart-cbt
+
+BigTable connector can be used as main outputs. A main output
+(common case) is expected to be massive and will be split into
+manageable chunks and processed in parallel. In the example below
+we created a list of rows then passed to the GeneratedDirectRows
+DoFn to set the Cells and then we call the WriteToBigtable to insert
+those generated rows in the table.
+
+  main_table = (p
+       | 'Generate Row Values' >> beam.Create(row_values)
+       | 'Generate Direct Rows' >> beam.ParDo(GenerateDirectRows())
+       | 'Write to BT' >> beam.ParDo(WriteToBigtable(beam_options)))
+"""
+from __future__ import absolute_import
+
+import apache_beam as beam
+from apache_beam.metrics import Metrics
+from apache_beam.transforms.display import DisplayDataItem
+
+try:
+  from google.cloud.bigtable import Client
+  from google.cloud.bigtable.batcher import MutationsBatcher
+except ImportError:
+  pass
+
+
+class WriteToBigtable(beam.DoFn):
+  """ Creates the connector can call and add_row to the batcher using each
+  row in beam pipe line
+
+  :type beam_options: class:`~bigtable_configuration.BigtableConfiguration`
+  :param beam_options: class `~bigtable_configuration.BigtableConfiguration`
+  """
+
+  def __init__(self, beam_options):
+    super(self.__class__, self).__init__(beam_options)
+    self.beam_options = beam_options
+    self.client = None
+    self.instance = None
+    self.table = None
+    self.batcher = None
+    self._app_profile_id = self.beam_options.app_profile_id
+    self.flush_count = self.beam_options.flush_count
+    self.max_row_bytes = self.beam_options.max_row_bytes
+    self.written = Metrics.counter(self.__class__, 'Written Row')
+
+  def start_bundle(self):
+    if self.beam_options.credentials is None:
+      self.client = Client(project=self.beam_options.project_id,
+                           admin=True)
+    else:
+      self.client = Client(project=self.beam_options.project_id,
+                           credentials=self.beam_options.credentials,
+                           admin=True)
+    self.instance = self.client.instance(self.beam_options.instance_id)
+    self.table = self.instance.table(self.beam_options.table_id,
+                                     self._app_profile_id)
+
+    self.batcher = MutationsBatcher(self.table, flush_count=self.flush_count,
+                                    max_row_bytes=self.max_row_bytes)
+    self.written = Metrics.counter(self.__class__, 'Written Row')
```

Comment:
Isn't `written` already created in the constructor?  This should only be created once.